### DISCUSSION TITLE
Invalid server version

### DISCUSSION THREAD

**Discussion by Ivan-B-V** [LsHallo/ets2-dedicated-convoy-server#16] [Q&A] [answered]
Hello. I tried to start dedicated server with docker. I have done this, but I have some issues.
The first one is such king of errors:
![Screenshot_3](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/56195207/4bdc7376-96f7-4365-915c-e3f228b849a7)
The second one is that I can't connect to the sever because "Invalid server version" error present.
![Screenshot_1](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/56195207/0b36ae9d-20e1-4418-9f78-99aa59e78447)
But I noticed that not only with my server this issue, with other similar issue.
![Screenshot_2](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/56195207/9cad264c-ec69-4f1a-8e87-2d97e36bc8eb)
Maybe somebody has solution for this or it's "SCS Software" problem and we need to wai till they solve this.
My docker-compose file below:
[docker-compose.txt](https://github.com/LsHallo/ets2-dedicated-convoy-server/files/14258050/docker-compose.txt)

**Comment by LsHallo** [comment, marked as answer]
The errors also appear for me. I think there is a problem with the server software right now. However they don't seem to affect the server much if at all.

Your game seems out of date. That's why no server seems to match. The server will update to the newest version on every startup if you don't explicitly disable it.
If you don't want to update your game you can take a look at the Steam beta versions of ETS and use the correct one that matches your game version using the beta environment variable of the server.

**Reply by Ivan-B-V** [reply]
Thanks for the solution. I set ETS_SERVER_BRANCH  to "public" and now it's working.

**Reply by Ivan-B-V** [reply]
Could you help me with issue that my server not appears in server list.
I created logon token and set it to variable, didn't use port mapping. You can see my docker-compose file.
Or it can be because I don't have static IP on my router?

**Reply by LsHallo** [reply]
You use network mode host in docker. That is the recommended (by me) way to ensure your server is visible.
If you are in a typical home network there may be 2 more layers that could prevent the ports being accessible.
1. The firewall of your server.
Please ensure 27015-27016 is allowed for UDP and TCP traffic.
2. The firewall of your router (port forwarding)
Also make sure you port forward 27015-27016 on your router to your server. Otherwise the packets will be dropped by your router.

**Reply by Ivan-B-V** [reply]
I'll check, thanks

**Reply by Ivan-B-V** [reply]
Thanks for your work. 
I commented "network_mode" and uncommented "ports". Now it works fine. 
Thanks for your docker image.
